### PR TITLE
provider/openstack: add in ability to autoscale based on cpu and lb traffic

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

**Comment by drmaas** [comment]
@spinnaker/reviewers ^^^

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy [hunk 1]
```diff
@@ -37,19 +38,31 @@ class ServerGroupParameters {
...
-      'image':image,
-      'internal_port':internalPort ? internalPort.toString() : null,
-      'max_size':maxSize ? maxSize.toString() : null,
-      'min_size':minSize ? minSize.toString() : null,
-      'desired_size':desiredSize ? desiredSize.toString() : null,
-      'network_id':networkId,
-      'subnet_id':subnetId,
-      'pool_id':poolId,
-      'security_groups':securityGroups ? securityGroups.join(',') : null
+      flavor:instanceType,
```
Comment (lwander) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy]: This seems to be the only place we don't have a space after the colon in a hashmap. Could you fix this for style?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy [hunk 2]
```diff
@@ -286,6 +287,14 @@ class OpenstackServerGroupCachingAgent extends AbstractOpenstackCachingAgent imp
       result.put('minSize', stack.parameters?.get('min_size') ?: 0)
       result.put('maxSize', stack.parameters?.get('max_size') ?: 0)
       result.put('desiredSize', stack.parameters?.get('desired_size') ?: 0)
+      result.put('autoscalingType', ServerGroupParameters.AutoscalingType.fromMeter(stack.parameters.get('autoscaling_type')))
+      ['up','down'].each {
+        Map<String, Object> scaler = [cooldown  : stack.parameters.get("scale${it}_cooldown"?.toString()),
```
Comment (lwander) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy]: I'm wondering why you have the `?.` operator on that string, I can't see it ever being `null`.

**Comment by lwander** [comment]
LGTM after comments

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy (hunk 2 above)
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy]: agreed, they will always have at least default template values.

**Comment by lwander** [comment]
Needs a rebase

**Comment by lwander** [comment]
Sorry! One more rebase...

**Pull request merged**